packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -1229,6 +1229,12 @@ mixin WidgetsBinding on BindingBase, ServicesBinding, SchedulerBinding, GestureB
   /// The `rootWidget` widget provided to this method must not already be
   /// wrapped in a [View].
   Widget wrapWithDefaultView(Widget rootWidget) {
+    assert(
```

Comment:
From my understanding, this error is caused by the user if they use our APIs incorrectly. Is `assert` the right thing to use here - I'd assume that should be for verifying our internal assumptions? Or should we use something like `throw FlutterError` instead?

Reply:
I guess the question is whether you want to terminate the app here in production or whether you want to keep running?

Reply:
Yeah I think `assert` is probably wrong - at least it won't work when assert is disabled such as in production. I suggest `assert` be only used on errors due to wrong internal implementation. For user errors like this, we should throw errors.

> whether you want to terminate the app here in production or whether you want to keep running

Theoretically we might be able to create a default view on such errors and continue, but I think we shouldn't do it, because any configuration without an implicit view is highly expert-level and users have to explicitly specify so to run into this error.

Reply:
Good point! I used assert to only throw while in development mode, but I agree it's silly, because this isn't recoverable and wouldn't work in production either. I'll `throw StateError` instead!